File: app/src/main/java/com/divercity/app/features/profile/selectgender/SelectGenderActivity.java
package com.divercity.app.features.profile.selectgender;

import android.content.Context;
import android.content.Intent;

import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.core.base.BaseFragmentActivity;

/**
 * Created by lucas on 17/10/2018.
 */

public class SelectGenderActivity extends BaseFragmentActivity {


    public static Intent getCallingIntent(Context context){
        Intent intent = new Intent(context, SelectGenderActivity.class);
        return intent;
    }

    @Override
    protected BaseFragment fragment() {
        return SelectGenderFragment.newInstance();
    }
}

File: app/src/main/java/com/divercity/app/features/onboarding/selectgender/SelectGenderActivity.kt
package com.divercity.app.features.onboarding.selectgender

import android.content.Context
import android.content.Intent
import com.divercity.app.core.base.BaseActivity
import com.divercity.app.core.base.BaseFragment

/**
 * Created by lucas on 17/10/2018.
 */

class SelectGenderActivity  : BaseActivity() {

    companion object {
        private const val INTENT_EXTRA_PARAM_PROGRESS = "progress"

        fun getCallingIntent(context: Context, progress : Int) : Intent {
            val intent = Intent(context, SelectGenderActivity::class.java)
            intent.putExtra(INTENT_EXTRA_PARAM_PROGRESS,progress)
            return intent
        }
    }

    override fun fragment(): BaseFragment = SelectGenderFragment.newInstance(
            intent.getIntExtra(INTENT_EXTRA_PARAM_PROGRESS, 0))
}
